src/gui/spatialArrow.js

```diff
@@ -0,0 +1,224 @@
+createNameSpace("realityEditor.gui.spatialArrow");
+
+import * as THREE from '../../thirdPartyCode/three/three.module.js';
+
+(function (exports) {
+    
+    let canvasContainer;
+    let canvas;
+    let ctx;
+    let screenW, screenH;
+    let screenRatio;
+
+    function initService() {
+        console.log('%c spatial arrow init!', 'color: green');
+        addCanvas();
+        resizeCanvas();
+        initCanvas();
+        update();
+    }
+
+    window.addEventListener('resize', () => {
+        resizeCanvas();
+        initCanvas();
+        update();
+    });
+    
+    let menuBarHeight;
+
+    function addCanvas() {
+        canvasContainer = document.createElement('div');
+        canvasContainer.className = 'arrow-canvas-container';
+        canvasContainer.style.position = 'absolute';
+        canvasContainer.style.top = '0';
+        canvasContainer.style.left = '0';
+        document.body.appendChild(canvasContainer);
+
+        canvas = document.createElement('canvas');
+        canvas.className = 'arrow-canvas';
+        canvas.style.position = 'absolute';
+        menuBarHeight = realityEditor.device.environment.variables.screenTopOffset;
+        canvas.style.top = `${menuBarHeight}px`;
+        canvas.style.left = '0';
+        canvas.style.zIndex = '3001';
+        canvasContainer.appendChild(canvas);
+
+        ctx = canvas.getContext("2d");
+    }
+
+    function resizeCanvas() {
+        if (canvas !== undefined) {
+            screenW = window.innerWidth;
+            screenH = window.innerHeight - menuBarHeight;
+            screenRatio = screenH / screenW;
+            canvas.width = screenW;
+            canvas.height = screenH;
+        }
+    }
+
+    const Clamp = (x, low, high) => {
```

Comment:
It's more standard to give functions a lowercase name so they're not confused with classes